tests/snuba/rules/conditions/test_event_frequency.py

```diff
@@ -1,248 +1,294 @@
+        data = {"interval": "1d", "value": 16}
+        self._run_test(data=data, minutes=1440, passes=False)
+
+    def test_one_week_with_events(self):
+        data = {"interval": "1w", "value": 6}
+        self._run_test(data=data, minutes=10080, passes=True, add_events=True)
+        data = {"interval": "1w", "value": 16}
+        self._run_test(data=data, minutes=10080, passes=False)
+
+    def test_thirty_days_with_events(self):
+        data = {"interval": "30d", "value": 6}
+        self._run_test(data=data, minutes=43200 - 1, passes=True, add_events=True)
+        data = {"interval": "30d", "value": 16}
+        self._run_test(data=data, minutes=43200 - 1, passes=False)
+
+    def test_one_minute_no_events(self):
+        data = {"interval": "1m", "value": 6}
+        self._run_test(data=data, minutes=1, passes=False)
+
+    def test_one_hour_no_events(self):
+        data = {"interval": "1h", "value": 6}
+        self._run_test(data=data, minutes=60, passes=False)
+
+    def test_one_day_no_events(self):
+        data = {"interval": "1d", "value": 6}
+        self._run_test(data=data, minutes=1440, passes=False)
+
+    def test_one_week_no_events(self):
+        data = {"interval": "1w", "value": 6}
+        self._run_test(data=data, minutes=10080, passes=False)
+
+    def test_thirty_days_no_events(self):
+        data = {"interval": "30d", "value": 6}
+        self._run_test(data=data, minutes=43200 - 1, passes=False)
+
+
+class EventFrequencyConditionTestCase(
+    FrequencyConditionMixin, StandardIntervalMixin, RuleTestCase, SnubaTestCase
+):
+    rule_cls = EventFrequencyCondition
+
+    def increment(self, event, count, environment=None, timestamp=None):
+        data = {
+            "fingerprint": event.data["fingerprint"],
+            "timestamp": iso_format(timestamp) if timestamp else iso_format(before_now(minutes=1)),
+        }
+        if environment:
+            data["environment"] = environment
+
+        for _ in range(count):
+            self.store_event(
+                data=data,
+                project_id=self.project.id,
+            )
+
+
+class EventUniqueUserFrequencyConditionTestCase(
+    FrequencyConditionMixin,
+    StandardIntervalMixin,
+    RuleTestCase,
+    SnubaTestCase,
+):
+    rule_cls = EventUniqueUserFrequencyCondition
+
+    def increment(self, event, count, environment=None, timestamp=None):
+        data = {
+            "fingerprint": event.data["fingerprint"],
+            "timestamp": iso_format(timestamp) if timestamp else iso_format(before_now(minutes=1)),
+        }
+        if environment:
+            data["environment"] = environment
+
+        for _ in range(count):
+            event_data = deepcopy(data)
+            event_data["user"] = {"id": uuid4().hex}
+            self.store_event(
+                data=event_data,
+                project_id=self.project.id,
+            )
+
+
+class EventFrequencyPercentConditionTestCase(
```

Comment:
These tests take quite a while to run, 45-50 seconds... wondering if we need to create 60 session buckets for all the tests?

Alternatively, I'm wondering if we can just create sessions/events in bulk. `store_session` actually puts the session into a list, so we could probably have an alternate version that just accepts all the sessions at once.

Reply:
Yeah I was actually thinking these tests were taking awhile. Part of that problem was introduced when we queried over the hour. I an making 60 sessions right now so that I get 1 session per minute on average and can make about half the events. Can speed this up a bit by making less sessions and customizing the tests a bit more and try that bulk method

Reply:
Running just `EventFrequencyPercentConditionTestCase` locally, I see the tests are about twice as fast by just implementing a bulk method (from about ~30->~15 seconds on average for me)

Reply:
Nice, it'd be good to introduce a bulk event creation method too, but I think that would take a bit more work.